subsys/net/ip/connection.c

```diff
@@ -505,6 +505,51 @@ static bool conn_are_end_points_valid(struct net_pkt *pkt,
 	return !(my_src_addr && (src_port == dst_port));
 }
 
+#if defined(CONFIG_NET_SOCKETS_PACKET)
+static enum net_verdict conn_raw_socket(struct net_pkt *pkt,
+					struct net_conn *conn)
+{
+	if (conn->flags & NET_CONN_LOCAL_ADDR_SET) {
+		struct net_if *pkt_iface = net_pkt_iface(pkt);
+		uint8_t proto = ETH_P_ALL;
+		struct sockaddr_ll *local;
+		struct net_pkt *raw_pkt;
+
+		local = (struct sockaddr_ll *)&conn->local_addr;
+
+		if (local->sll_ifindex !=
+		    net_if_get_by_iface(pkt_iface)) {
+			return NET_CONTINUE;
+		}
+
+		NET_DBG("[%p] raw match found cb %p ud %p",
+			conn, conn->cb,	conn->user_data);
+
+		raw_pkt = net_pkt_clone(pkt, CLONE_TIMEOUT);
+		if (!raw_pkt) {
+			net_stats_update_per_proto_drop(pkt_iface, proto);
+
+			return NET_DROP;
+		}
+
+		if (conn->cb(conn, raw_pkt, NULL,
+			     NULL, conn->user_data) ==
```

Comment:
`NET_DROP) {` could be placed to this line, it does not look too long

Reply:
Fixed. I've pushed new version of those patches.